src/types.jl

```diff
@@ -149,12 +507,552 @@ for Model in [:NeuralNetworkRegressor, :MultitargetNeuralNetworkRegressor]
+  function which can be called in the form `loss(yhat, y)`. Possible loss functions are
+  listed in [the Flux loss function
+  documentation](https://fluxml.ai/Flux.jl/stable/models/losses/). For a regression task,
+  the most natural loss functions are:
+    - `Flux.mse`
+    - `Flux.mae`
+    - `Flux.msle`
+    - `Flux.huber_loss`
+- `epochs::Int=10`: The number of epochs to train for. Typically, one epoch represents
+  one pass through the entirety of the training dataset.
+- `batch_size::Int=1`: The batch size to be used for training. The batch size represents
+  the number of samples per update of the networks weights. Typcally, batch size should be
+  somewhere between 8 and 512. Smaller batch sizes lead to noisier training loss curves,
+  while larger batch sizes lead towards smoother training loss curves. In general, it is a
+  good idea to pick one fairly large batch size (e.g. 32, 64, 128), and stick with it, and
+  only tune the learning rate. In most literature, batch size is set in powers of twos,
+  but this is fairly arbitrary.
+- `lambda::Float64=0`: The stregth of the regularization used during training. Can be
+  any value  in the range `[0, ∞)`.
+- `alpha::Float64=0`: The L2/L1 mix of regularization, in the range `[0, 1]`. A value of
+  0 represents L2 regularization, and a value of 1 represents L1 regularization.
+- `rng::Union{AbstractRNG, Int64}`: The random number generator/seed used during
+  training.
+- `optimizer_changes_trigger_retraining::Bool=false`: Defines what happens when fitting
+  a machine if the associated optimiser has changed. If true, the associated machine will
+  retrain from scratch on `fit`, otherwise it will not.
+- `acceleration::AbstractResource=CPU1()`: Defines on what hardware training is done.
+  For Training on GPU, use `CudaLibs()`, otherwise defaults to `CPU`()`.
+- `finaliser=Flux.softmax`: The final activation function of the neural network.
+Defaults to `Flux.softmax`. For a regression task, reasonable alternatives include
+`Flux.sigmoid` and the identity function (otherwise known as "linear activation").
+
+# Operations
+
+- `predict(mach, Xnew)`: return predictions of the target given new
+  features `Xnew` having the same Scitype as `X` above. Predictions are
+  deterministic.
+
+
+# Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+- `chain`: The trained "chain", or series of layers, functions, and activations which
+  make up the neural network.
+
+
+# Report
+
+The fields of `report(mach)` are:
+
+- `training_losses`: The history of training losses, a vector containing the history of
+  all the losses during training. The first element of the vector is the initial
+  penalized loss. After the first element, the nth element corresponds to the loss of
+  epoch n-1.
+
+# Examples
+
+In this example we build a regression model using a toy dataset.
+```julia
+using MLJ
+using MLJFlux
+using Flux
+using Plots
+using MLJBase: augment_X
+```
```

Comment:
We need to tweak the example to avoid calling `augment_X` (https://github.com/JuliaAI/MLJBase.jl/pull/780)